# Diabetes Dataset - EDA

## Opening and First look

In [1]:
# Imports
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
dataset = pd.read_csv('diabetes_unclean.csv')

In [22]:
dataset

ID  No_Pation Gender   AGE  Urea     Cr  HbA1c  Chol   TG  HDL  LDL  \
0     502      17975      F  50.0   4.7   46.0    4.9   4.2  0.9  2.4  1.4   
1     735      34221      M  26.0   4.5   62.0    4.9   3.7  1.4  1.1  2.1   
2     420      47975      F  50.0   4.7   46.0    4.9   4.2  0.9  2.4  1.4   
3     680      87656      F  50.0   4.7   46.0    4.9   4.2  0.9  2.4  1.4   
4     504      34223      M  33.0   7.1   46.0    4.9   4.9  1.0  0.8  2.0   
...   ...        ...    ...   ...   ...    ...    ...   ...  ...  ...  ...   
1000  185     454316      M  64.0   8.8  106.0    8.5   5.9  2.1  1.2  4.0   
1002  188     454316      F  75.0  10.3  113.0    8.6   4.2  1.6  0.9  2.6   
1003  189     454316      M  58.0   4.0   55.0    7.9   4.9  2.0  1.2  1.4   
1007  194     454316      F  57.0   4.1   70.0    9.3   5.3  3.3  1.0  1.4   
1008  195       4543      f  55.0   4.1   34.0   13.9   5.4  1.6  1.6  3.1   

      VLDL   BMI CLASS  
0      0.5  24.0     N  
1      0.6  23.0     N  
2      0.5  24.0     N  
3      0.5  24.0     N  
4      0.4  21.0     N  
...    ...   ...   ...  
1000   1.2  32.0    Y   
1002   0.7  32.0    Y   
1003   1.1  35.0    Y   
1007   1.3  29.0    Y   
1008   0.7  33.0    Y   

[994 rows x 14 columns]

In [4]:
dataset.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1009 entries, 0 to 1008
Data columns (total 14 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   ID         1009 non-null   int64  
 1   No_Pation  1009 non-null   int64  
 2   Gender     1009 non-null   object 
 3   AGE        1008 non-null   float64
 4   Urea       1008 non-null   float64
 5   Cr         1007 non-null   float64
 6   HbA1c      1006 non-null   float64
 7   Chol       1007 non-null   float64
 8   TG         1007 non-null   float64
 9   HDL        1008 non-null   float64
 10  LDL        1007 non-null   float64
 11  VLDL       1008 non-null   float64
 12  BMI        1009 non-null   float64
 13  CLASS      1009 non-null   object 
dtypes: float64(10), int64(2), object(2)
memory usage: 110.5+ KB


In [5]:
dataset.describe()

ID     No_Pation          AGE         Urea           Cr  \
count  1009.000000  1.009000e+03  1008.000000  1008.000000  1007.000000   
mean    339.161546  2.717448e+05    53.620040     5.131094    68.973188   
std     239.738169  3.365681e+06     8.740975     2.931136    59.813297   
min       1.000000  1.230000e+02    25.000000     0.500000     6.000000   
25%     127.000000  2.406500e+04    51.000000     3.700000    48.000000   
50%     296.000000  3.439900e+04    55.000000     4.600000    60.000000   
75%     548.000000  4.539000e+04    59.000000     5.700000    73.000000   
max     800.000000  7.543566e+07    79.000000    38.900000   800.000000   

             HbA1c         Chol           TG          HDL          LDL  \
count  1006.000000  1007.000000  1007.000000  1008.000000  1007.000000   
mean      8.284155     4.863873     2.348769     1.204216     2.610119   
std       2.533576     1.297326     1.397487     0.658158     1.116095   
min       0.900000     0.000000     0.300000     0.200000     0.300000   
25%       6.500000     4.000000     1.500000     0.900000     1.800000   
50%       8.000000     4.800000     2.000000     1.100000     2.500000   
75%      10.200000     5.600000     2.900000     1.300000     3.300000   
max      16.000000    10.300000    13.800000     9.900000     9.900000   

              VLDL          BMI  
count  1008.000000  1009.000000  
mean      1.850595    29.589911  
std       3.649859     4.946246  
min       0.100000    19.000000  
25%       0.700000    26.000000  
50%       0.900000    30.000000  
75%       1.500000    33.000000  
max      35.000000    47.750000

# Data Cleaning and Preprocessing
- Backup the original dataset
- Check for Null values
- Conversion to numerical values or one-hot encoding of the 'object' or 'string' types
- Check for outliers 

In [13]:
ds_backup = dataset.copy()
ds_backup.to_csv('diabetes_unclean_backup.csv', index=False)

In [10]:
dataset.isnull().sum()

ID           0
No_Pation    0
Gender       0
AGE          1
Urea         1
Cr           2
HbA1c        3
Chol         2
TG           2
HDL          1
LDL          2
VLDL         1
BMI          0
CLASS        0
dtype: int64

### Null values 
Since the total entries of the dataset are 1009 and the Null values are around the 2% of the dataset it is possible to remove them (no fill required)


In [14]:
dataset.dropna(inplace=True)

### Conversion to numerical values

In [20]:
dataset.dtypes

ID             int64
No_Pation      int64
Gender        object
AGE          float64
Urea         float64
Cr           float64
HbA1c        float64
Chol         float64
TG           float64
HDL          float64
LDL          float64
VLDL         float64
BMI          float64
CLASS         object
dtype: object

Columns to be converted or encoded:
- Gender -> simple encoding [0 = 'M', 1 = 'F']
- Class -> simple encoding [0 = non-positive, 1 = positive]


In [26]:
dataset['Gender'].unique()

array(['F', 'M', 'f'], dtype=object)

As there are inconsistencies in the gender column 

In [ ]:

print(dataset['CLASS'].unique())

array(['N', 'N ', 'P', 'Y', 'Y '], dtype=object)